internal/openstack/client.go

```diff
@@ -144,8 +145,9 @@ func NewClient(cfg *AuthOpts) (*OpenStackClient, error) {
 	}
 
 	return &OpenStackClient{
-		provider: provider,
-		region:   cfg.Region,
+		provider:  provider,
+		region:    cfg.Region,
+		projectID: cfg.ProjectID,
```

Comment:
In the common uses in yaook/k8s, projectID will be nil: https://gitlab.com/yaook/k8s/-/blob/devel/k8s-base/roles/ch-k8s-lbaas-controller/templates/controller-config.toml?ref_type=heads#L9-19

So in cases where the project ID is nil, we need to resolve it at startup using the project name and project domain id.

Reply:
I implemented this, but it's still untested.